File: src/java/_08final/mvc/model/NukeFloater.java
package _08final.mvc.model;

import _08final.mvc.controller.CommandCenter;
import _08final.mvc.controller.Game;
import _08final.mvc.controller.SoundLoader;

import java.awt.*;
import java.util.LinkedList;

public class NukeFloater extends Floater {

	//spawn every 12 seconds
	public static final int SPAWN_NUKE_FLOATER = Game.FRAMES_PER_SECOND * 12;
	//frames before this floater expires of natural mortality
	public static final int EXPIRY = 120;
	public NukeFloater() {
		setColor(Color.YELLOW);
		setExpiry(EXPIRY);
	}

	@Override
	public void removeFromGame(LinkedList<Movable> list) {
		super.removeFromGame(list);
		//if getExpiry() > 0, then this remove was the result of a collision, rather than natural mortality
		if (getExpiry() > 0) {
			SoundLoader.playSound("nuke-up.wav");
			CommandCenter.getInstance().getFalcon().setNukeMeter(Falcon.MAX_NUKE);
		}

	}


}

File: src/python/_08final/mvc/model/NukeFloater.py

from mvc.controller.CommandCenter import CommandCenter
from mvc.controller.SoundLoader import SoundLoader
from mvc.model.Falcon import Falcon
from mvc.model.Floater import Floater
from mvc.model.prime.Color import Color


class NukeFloater(Floater):
    # spawn every 14 seconds (in java Game.FRAMES_PER_SECOND * 12)
    SPAWN_NUKE_FLOATER = 350
    # frames before this floater expires of natural mortality
    EXPIRY = 350

    def __init__(self):
        super().__init__()
        #yellow
        self.color = Color.YELLOW
        self.expiry = NukeFloater.EXPIRY



    def removeFromGame(self, list):
        super().removeFromGame(list)
        # if expiry > 0, then this remove was the result of a collision w/Falcon, and not natural mortality.
        if (self.expiry > 0):
            CommandCenter.getInstance().falcon.nukeMeter = Falcon.MAX_NUKE
            SoundLoader.playSound("nuke-up.wav")